cloud/cloud_env_impl.cc

```diff
@@ -851,11 +851,15 @@ void CloudEnvImpl::StopPurger() {
 }
 
 Status CloudEnvImpl::LoadLocalCloudManifest(const std::string& dbname) {
+  return LoadLocalCloudManifest(dbname, cloud_env_options.cookie_on_open);
+}
+
+Status CloudEnvImpl::LoadLocalCloudManifest(const std::string& dbname, const std::string& cookie) {
```

Comment:
When do you ever create the cloud manifest with the “old” cookie?

I feel that you should
- load initial cloud manifest with old cookie
- write new cloud manifest with new cookie
- set `options_.cookie_on_open = options_.new_cookie_on_open`, after this point we only work in the “new” world